backend/common.go

```diff
@@ -110,6 +115,7 @@ func (s *DataSourceInstanceSettings) HTTPClientOptions() (httpclient.Options, er
 	opts := httpSettings.HTTPClientOptions()
 	opts.Labels["datasource_name"] = s.Name
 	opts.Labels["datasource_uid"] = s.UID
+	opts.Labels["datasource_type"] = s.Type
```

Comment:
@marefr `PluginContext` isn't available here, so it seems like the other option would be to change the signature of `func (s *DataSourceInstanceSettings) HTTPClientOptions() (httpclient.Options, error)` to accept a plugin ID. is that what you were thinking?

if adding `Type` seems like a better option, would adding plugin ID in as a second argument here be enough to avoid changing the protobuf? https://github.com/grafana/grafana-plugin-sdk-go/blob/6d1d0f5edbb6365994a49d4326a2aefa407fcb87/backend/convert_from_protobuf.go#L80

Reply:
> `PluginContext` isn't available here

I know. The idea would be to populate the public DataSourceInstance contract with `Type` from `PluginContext.PluginID` when converting from protobuf https://github.com/grafana/grafana-plugin-sdk-go/blob/e757e0dd5bde9f06bb0431523e2942bebb15e2b8/backend/convert_from_protobuf.go#L52

Reply:
👍 yep, we are on the same page i think. updated in [c3d044d](https://github.com/grafana/grafana-plugin-sdk-go/pull/490/commits/c3d044da841ddb28af9b796e977ba8b236fc1a60)